google/cloud/internal/opentelemetry.cc

```diff
@@ -36,24 +36,28 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> GetTracer(
 opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpanImpl(
     opentelemetry::nostd::string_view name,
     opentelemetry::common::KeyValueIterable const& attributes,
-    opentelemetry::trace::SpanContextKeyValueIterable const& links) {
-  opentelemetry::trace::StartSpanOptions options;
-  options.kind = opentelemetry::trace::SpanKind::kClient;
+    opentelemetry::trace::SpanContextKeyValueIterable const& links,
+    absl::optional<opentelemetry::trace::StartSpanOptions> const& options) {
+  opentelemetry::trace::StartSpanOptions default_options;
+  default_options.kind = opentelemetry::trace::SpanKind::kClient;
   return GetTracer(CurrentOptions())
-      ->StartSpan(name, attributes, links, options);
+      ->StartSpan(name, attributes, links,
+                  (options.has_value()) ? options.value() : default_options);
```

Comment:
Q. Could this be done at a higher level?  For example, rather than passing `nullopt`, pass `DefaultStartSpanOptions()`, and then get rid of `optional`?

Reply:
I don't see a way out of using an optional.

99% of the spans are kClient, but there are 2 spans for Pub/Sub that are kProducer and kConsumer.

I need some way to indicate I don't want those to be overwritten.

Reply:
Did the discussion about `optional<StartSpanOptions> options = nullopt` clarify this?

Even if we stuck with defaulted arguments, why would `StartSpanOptions options = DefaultStartSpanOptions()` not be a better choice than adding an `optional` wrapper?

Reply:
Yes. changed.